File: 3179-find-the-n-th-value-after-k-seconds/3179-find-the-n-th-value-after-k-seconds.py
class Solution:
    def valueAfterKSeconds(self, n: int, k: int) -> int:
        MOD = 1_000_000_007;
        dp = [1] * n
        
        for i in range(1, k+1):
            for j in range(1, n):
                dp[j] = (dp[j-1] + dp[j]) % MOD
        
        return dp[n-1]

File: 3179-find-the-n-th-value-after-k-seconds/3179-find-the-n-th-value-after-k-seconds.java
class Solution {
    public int valueAfterKSeconds(int n, int k) {
        var MOD = 1_000_000_007;
        var dp = new int[n];
        for(var i=0; i<n; i++) dp[i] = 1;
        
        for(var i=1; i<=k; i++)
            for(var j=1; j<n; j++)
                dp[j] = (dp[j-1] + dp[j]) % MOD;
        
        return dp[n-1];
    }
}